Product Page.

arichylton opened the pull request:
Product page that uses image swapper component and product information component to render contents.  Error handling for when id in URL params doesn't exist in product database. 

![image](https://github.com/4human-org/murphy-ecommerce-frontend/assets/45323985/63148a28-4418-41f1-87fb-ead853f183c4)

Product data is fetched from DummyData.js by using FetchProductById.js in utils.

All information for product will be updated when back-end integration happens. 
Add to cart functionality is not in place yet either. Need a global checkout list state variable since the only way to access it now is to be a child of CartPage.jsx 

Will continue development of this when back-end integration happens.

Connor-Skudlarek committed:
Navbar component now PascalCase, "use client", and .jsx

Connor-Skudlarek committed:
Home page and layout now .jsx, home page no longer showing cart page

The home page will now show the actual home page again.

arichylton committed:
Added page routing based on id params in the url, image swapper component and product information component for each product and error handling if the product in url is not found.

Connor-Skudlarek commented on app/product/ProductInformation.jsx:
```diff
@@ -0,0 +1,94 @@
+    setActiveTab(tabId);
+  };
+
+  // Will need global state or push to database to effectively add to cart so current functionality is not implemented
```
Noted. Will look into this more when backend is ready.

Connor-Skudlarek replied:
@kailahulse can you add this to your notes on your end, too?

Connor-Skudlarek reviewed (changes requested):
Nice work on this! Only a small change is needed.

With App Router it is good practice to make sure components are in the components directory, separate from the route itself.

If the component is specific to this route (like these look to be), we can make a components/product directory to hold them. It will help with navigating the codebase as we grow (especially for new people) to find where different components are located.

Otherwise looks really good. I appreciate the short comments.

Connor-Skudlarek committed:
Create results page and move results components to component directory, add Suspense boundaries

Moves components to correct directories rather than clog up the routing directory in /app. Also fixes some Suspense boundaries errors cropping up during "npm run build".

Connor-Skudlarek committed:
Move cart page to own page, corrects minor routing, naming, and CSS issues

Connor-Skudlarek committed:
Move Suspense import into same line

Connor-Skudlarek committed:
Merge branch 'routing-and-naming-bugs'

Connor-Skudlarek committed:
Merge branch 'main' into productPage

Connor-Skudlarek committed:
Move everything to separate directory to not clog App Router, #12

Connor-Skudlarek reviewed (approved):
Files now moved to correct directory. Good to go.
